tests/fips/fips_setup.pm

```diff
@@ -39,8 +39,11 @@ sub enable_fips {
         validate_script_output("fips-mode-setup --check", sub { m/FIPS mode is enabled\.\n.*\nThe current crypto policy \(FIPS\) is based on the FIPS policy\./ });
     } else {
         if (is_transactional) {
-            change_grub_config('=\"[^\"]*', '& fips=1 ', 'GRUB_CMDLINE_LINUX_DEFAULT');
-            trup_call('--continue grub.cfg');
+            # on SL Micro 6.0+ we only need to reboot, no need to manually change grub.
+            if (is_sle_micro('<6.0')) {
+                change_grub_config('=\"[^\"]*', '& fips=1 ', 'GRUB_CMDLINE_LINUX_DEFAULT');
+                trup_call('--continue grub.cfg');
+            }
```

Comment:
Does this have to be a nested condition? What about something like this:

```
if (is_transactional && is_sle_micro('<6.0')) {
}
```

Reply:
actually, I can even remove the `is_transactional`

Reply:
not really, because then with sle micro 6.0 it would execute the else branch which would be wrong

Reply:
but I can simplify it, gimme a sec